### PR TITLE
NMS-12212: Show configured rule groups

### PR DESCRIPTION
**Pull request by patrick-schweizer**
* Jira: https://issues.opennms.org/browse/NMS-12212

# Test Instructions:
* Manually add new rule group into database, e.g. `INSERT INTO classification_groups ("id","name", "description", "readonly", "enabled", "priority") VALUES ('5','group5', 'another user defined group', false, true, 40);`
* group should show up in ui
* add rule
* edit rule and select different group: rule should be moved to different group. Positions in both group should have been rearranged

### CODE REVIEW

**Comment by mvrueden** [comment]
I tried this and it works. 

Some things I noticed, which probably needs some offline discussions: 

Do we want to allow to create read only groups?
If we do allow, this means we should also allow for editing those (meaning make them editable).
But for the System-defined rules we probably don't want to make them editable.
So probably best to not allow to create read-only groups at all. -> let's talk about this

The group name must be unique. Can you add a test to the rest endpoint verifying that when posting a group with an already existing name, you get a `400 Bad Request: { context: 'name', message: 'A group with name "bla" already exists }`.

The definition of a rule must be unique within a group.
If I were to move a rule to a group where a rule with the same definition already exists, the UI only reports `A rule with this definition already exists`. Maybe just a "move" action is suited better for this use case instead of adding it to the edit dialogue. -> Let's talk about this

When adding a new rule to a group, the drop down allows to change the group while creating it.
Not sure if we want to allow this, as the user already selected the group in order to add a rule. -> Let's talk about this

File: features/flows/rest/api/src/main/java/org/opennms/netmgt/flows/rest/classification/GroupDTOBuilder.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
+/*******************************************************************************
+ * This file is part of OpenNMS(R).
+ *
+ * Copyright (C) 2018-2018 The OpenNMS Group, Inc.
```
Comment (mvrueden) [inline features/flows/rest/api/src/main/java/org/opennms/netmgt/flows/rest/classification/GroupDTOBuilder.java]: Should be `2019-2019`.

File: features/flows/rest/api/src/main/java/org/opennms/netmgt/flows/rest/classification/GroupDTOBuilder.java [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+ * OpenNMS(R) is Copyright (C) 1999-2018 The OpenNMS Group, Inc.
```
Comment (mvrueden) [inline features/flows/rest/api/src/main/java/org/opennms/netmgt/flows/rest/classification/GroupDTOBuilder.java]: Should be `1999-2019`.

File: features/flows/rest/api/src/main/java/org/opennms/netmgt/flows/rest/classification/GroupDTOBuilder.java (hunk 1 above)
Comment (patrick-schweizer) [reply features/flows/rest/api/src/main/java/org/opennms/netmgt/flows/rest/classification/GroupDTOBuilder.java]: fixed

**Comment by j-white** [comment]
This was closed because the `features/drift2` branch was deleted. Please re-open and retarget to `release-25.0.0`.

**Comment by patrick-schweizer** [comment]
replaced by: https://github.com/OpenNMS/opennms/pull/2661

**Pull request closed without merging**